static/js/jira-server-url.js

```diff
@@ -55,14 +56,61 @@ const setErrorMessage = error => {
 	$("#gheServerURL").addClass("has-error");
 };
 
-/**
- * Hides the error messages
- */
 const hideErrorMessage = () => {
 	$("#gheServerURLError").hide();
 	$("#gheServerURL").removeClass("has-error");
 };
 
+const activeRequest = () => {
+	$("#gheServerBtnText").hide();
+	$("#gheServerBtnSpinner").show();
+};
+
+const requestFailed = () => {
+	$("#gheServerBtnText").show();
+	$("#gheServerBtnSpinner").hide();
+};
+
+const checkForOrCreateGitHubApp = (gheServerURL) => {
+	$.ajax({
+		type: "POST",
+		url: "/jira/app-creation",
+		data: {
+			gheServerURL
+		},
+		success: function(data) {
+			const pagePath =  data.moduleKey;
+			AP.navigator.go(
+				"addonmodule",
+				{
+					moduleKey: pagePath
+				}
+			);
+		},
+		error: function(err) {
+			console.error(`Failed to retrieve GH app data. ${err}`)
+			// TODO - build and render error component
+		}
+	});
+}
+
+const verifyGitHubServerUrl = (gheServerURL) => {
+	$.ajax({
+		type: "POST",
+		url: "/jira/verify-server-url",
+		data: {
+			gheServerURL
+		},
+		success: function() {
+			checkForOrCreateGitHubApp(gheServerURL);
```

Comment:
I'm not sure I understand this.  We check the URL first, then just do another ajax call to create the server app?  Don't we already know the server url by now as part of the flow (the previous page)?

Reply:
The previous page is the following, so no knowledge of the url prior to this page.

<img width="541" alt="Screen Shot 2022-06-30 at 11 50 52 am" src="https://user-images.githubusercontent.com/37155488/176575518-3e74b5e7-d1bb-4749-be64-e4c1bd6a39ee.png">

The first request verifies that we hit the provided url and get a 200 response. If that fails, we render the error component (coming soon) but if it succeeds we then query the db to check for exisiting GH apps. Can condense into one request that does both the url verification and db query checks if you think that makes more sense.